### ISSUE TITLE
Idea: BFGminer as CoinSwitch?

### ISSUE THREAD

**Issue by ghost** [luke-jr/bfgminer#665]
Hi luke-jr. Thank you for your hard work on bfgminer and your great support here..
just a quick question: do you have any ideas ( or know any people working on ) how to use bfgminer as a profit-switch? ( like a pool, but one payout adress is enough )

my idea: i would connect bfgminer to different coin daemons and start the stratum proxy port. then i connect several miners on that stratum port. then i could do some magic ( external software ) to determine the most profitable coin. then i could call the bfgminer-api and say "switch to the different daemon".

is this possible? and how fast would you rate that switching? i am coming from the pool scene ( written in node js ), but are wondering if a pure c program wouldn't be faster? ( i would really like to go down to 100ms )  i mean it seems that there is everything present i need? only problem would be to optimize that switching..( extranonce1 etc ) as i never worked in C, i would love to hear your opinion on that..

thank you!

**Comment by luke-jr** [comment]
Yes, that would work, but you'd probably need to limit changes to once per new block, and/or use the multi-goal stratum extensions (which only BFGMiner supports AFAIK). Be careful to avoid giving the same jobs twice (only switching once per block also should do this).
